File: frontend/src/types/execution.ts
export type ExecutionStatus = 'pending' | 'running' | 'passed' | 'failed' | 'timeout' | 'cancelled';
export type TriggerType = 'manual' | 'scheduled';

export interface ExecutionSummary {
  id: number;
  test_id: number;
  schedule_id?: number | null;
  status: ExecutionStatus;
  started_at?: string | null;
  finished_at?: string | null;
  duration?: number | null;
  exit_code?: number | null;
  trigger_type: TriggerType;
  created_at: string;
}

export interface ExecutionDetail extends ExecutionSummary {
  stdout?: string | null;
  stderr?: string | null;
  test_name?: string | null;
  test_framework?: string | null;
  schedule_expression?: string | null;
}

export interface ExecutionListResponse {
  items: ExecutionSummary[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface ExecutionStats {
  total_executions: number;
  passed: number;
  failed: number;
  timeout: number;
  cancelled: number;
  pending: number;
  running: number;
  success_rate: number;
  average_duration: number;
}

File: backend/app/schemas/execution.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, ConfigDict, Field
from app.models.enums import ExecutionStatus, TriggerType


class ExecutionCreate(BaseModel):
    test_id: int = Field(..., description="ID of the test script to execute")


class ExecutionResponse(BaseModel):
    id: int
    test_id: int
    schedule_id: Optional[int] = None
    status: ExecutionStatus
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
    duration: Optional[float] = None
    exit_code: Optional[int] = None
    stdout: Optional[str] = None
    stderr: Optional[str] = None
    trigger_type: TriggerType
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ExecutionListResponse(BaseModel):
    items: List[ExecutionResponse]
    total: int
    page: int
    page_size: int
    total_pages: int
